core/accounting.go

```diff
@@ -0,0 +1,73 @@
+package core
+
+import (
+	"math/big"
+	"sync"
+	"time"
+)
+
+// Use transcodeLoopTimeout as TTL for a ManifestID's credit balance
+// We should prevent the use of global state if they are not constants
+
+// Accounts holds credit balances on a per-stream basis
+type Accounts struct {
+	Balances map[ManifestID]balance
+	Mtx      sync.RWMutex
+}
+
+type balance struct {
+	LastUpdate time.Time // Unix time since last update
+	Amount     *big.Rat  // Balance represented as a big.Rat
+}
+
+// Credit adds an an amount to the balance for a ManifestID
+func (a *Accounts) Credit(id ManifestID, amount *big.Rat) {
+	a.Mtx.Lock()
+	defer a.Mtx.Unlock()
+	if a.Balances[id].Amount == nil {
+		a.Balances[id] = balance{Amount: big.NewRat(0, 1)}
+	}
+	a.Balances[id] = balance{
+		LastUpdate: time.Now(),
+		Amount:     new(big.Rat).Add(a.Balances[id].Amount, amount),
+	}
+}
+
+// Debit substracts an mount from the balance for a ManifestID
```

Comment:
should be "an amount" , rather than "an mount" 🏇

Reply:
fixed in https://github.com/livepeer/go-livepeer/pull/956/commits/9a31f394cc058ad90f4f1566c6f5880490ef861d